File: AD-HOC/1521/main.dart
import 'dart:io';

int find(List<int> students, int k) {
  int student = students[k - 1];
  return (student == k) ? k : find(students, student);
}

void main() {
  while (int.parse(stdin.readLineSync()!.trim()) > 0) {
    List<int> students = stdin.readLineSync()!.trim().split(' ').map(int.parse).toList();
    int k = int.parse(stdin.readLineSync()!.trim());
    print(find(students, k));
  }
}

File: AD-HOC/1521/main.c
#include <stdio.h>

int find(int *students, int k) {
    int student = students[k - 1];
    return (student == k) ? k : find(students, student);
}

int main() {
    int n, k;
    for (scanf("%d", &n); n > 0; scanf("%d", &n)) {
        int students[n];
        for (int student = 0; student < n; student++) scanf("%d", &students[student]);
        scanf("%d", &k);
        printf("%d\n", find(students, k));
    }
    return 0;
}
